### DISCUSSION TITLE
How is this better than ri?

### DISCUSSION THREAD

**Discussion by Nakilon** [skatkov/rdoc-markdown#32] [General]
Considering this:

> ### Motivation
> I'm trying to depend less on software with GUI, instead using software that could be used through console

**Comment by skatkov** [comment]
@Nakilon This question probably belongs more in discussions. :)

But I will attempt to answer as good as possible, please consider that this is my opinion, if `ri` is good enough for you—then that's great. But I tried `ri` and `pry-doc` before starting this project and was very frustrated with experience.

These are the issues from my memory, they might not be up-to-date right now or I could be forgetting something:
- `ri` worked pretty stably for me for anything related to stdlib, but I had issues picking up gem docs with it. 
-  for gems, I had to use `pry-docs`, but it worked for me only if 'pry-doc' gem was part of a bundle. But somehow `pry-doc` had issues picking up stdlib classes.
- it took quite a lot of time and space to generate documentation for `ri` to use.  In some cases, I had to modify `.gemrc` to include docs for gems and I haven't found a way to do so selectively for gems.
- search had a lot of feature to be desired -- only class/method names could be used, fuzzy matching was not great, description was not indexed.
- Readability of documentation could be improved


I decided to build this gem to generate "markdown and search index". In my mind, this approach seems easier:
- markdown is well-supported by other software, It already works with software like [glow](https://github.com/charmbracelet/glow).
- CSS styling could be applied to improve readability.
- Markdown could be distributed to any device, that doesn't have ruby on it. I could potentially even ssh into a server to review documentation (or have a wearable device with access to docs) 

I'm also building a TUI client on top of this library—I call it [posh ](https://www.poshtui.com/). I plan to solve all the issues I had with `ri` and `pry-doc` with this combo.

**Comment by Nakilon** [comment]
Thanks for the explanation!
P.S.: there is no Discussions tab:
![image](https://user-images.githubusercontent.com/2870363/202902522-2c0dc5ea-d009-4381-bc9c-eccaee492609.png)

**Reply by skatkov** [reply]
@Nakilon  I was unaware that discussions were only enabled for maintainers. But I have enabled it for everyone and also moved this issue to discussions.

If your experience with `ri` was better than mine, I'd appreciate it if you'd share  it.
